src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/OpDelegationStrategy.cs

```diff
@@ -77,27 +77,28 @@ private bool IsSupportedNode(TexlNode node, OperationCapabilityMetadata metadata
 
-            if (!binding.IsRowScope(node))
+            // Check whether this is -
+            //  1) in operator delegation and
+            //  2) it is verifying if RHS node is supported and
+            //  3) it is not an async node and
+            //  4) it is a single column table and
+            //  5) metadata belongs to cds datasource that supports delegation of CdsIn
+            // If this check fails, verify if it is simply a valid node..
+            // Eg of valid delegation functions -
+            // Filter(Accounts, 'Account Name' in ["Foo", Bar"]) - Direct table use
+            // Set(Names, ["Foo", Bar"]); Filter(Accounts, 'Account Name' in Names) - Using variable of type table
+            // ClearCollect(Names, Accounts); Filter(Accounts, 'Account Name' in Names.'Account Name') - using column from collection.
+            // This won't be delegated - Filter(Accounts, 'Account Name' in Accounts.'Account Name') as Accounts.'Account Name' is async.
+            if (binding.Document.Properties.EnabledFeatures.IsEnhancedDelegationEnabled &&
+                isRHSNode && (opDelStrategy as BinaryOpDelegationStrategy)?.Op == BinaryOp.In && !binding.IsAsync(node) && binding.GetType(node).IsTable && binding.GetType(node).IsColumn &&
+                metadata.IsDelegationSupportedByTable(DelegationCapability.CdsIn))
             {
-                // Check whether this is -
-                //  1) in operator delegation and
-                //  2) it is verifying if RHS node is supported and
-                //  3) it is not an async node and
-                //  4) it is a single column table and
-                //  5) metadata belongs to cds datasource that supports delegation of CdsIn
-                // If this check fails, verify if it is simply a valid node..
-                // Eg of valid delegation functions -
-                // Filter(Accounts, 'Account Name' in ["Foo", Bar"]) - Direct table use
-                // Set(Names, ["Foo", Bar"]); Filter(Accounts, 'Account Name' in Names) - Using variable of type table
-                // ClearCollect(Names, Accounts); Filter(Accounts, 'Account Name' in Names.'Account Name') - using column from collection.
-                // This won't be delegated - Filter(Accounts, 'Account Name' in Accounts.'Account Name') as Accounts.'Account Name' is async.
-                if ((binding.Document.Properties.EnabledFeatures.IsEnhancedDelegationEnabled &&
-                    isRHSNode && (opDelStrategy as BinaryOpDelegationStrategy)?.Op == BinaryOp.In && !binding.IsAsync(node) && binding.GetType(node).IsTable && binding.GetType(node).IsColumn &&
-                    metadata.IsDelegationSupportedByTable(DelegationCapability.CdsIn)) ||
-                    IsValidNode(node, binding))
-                {
-                    return true;
-                }
+                return true;
+            }
+
+            if (!binding.IsRowScope(node) && IsValidNode(node, binding))
```

Comment:
>!binding.IsRowScope(node) [](http://example.com/codeflow?start=16&length=25)

What is the reason for moving this check below? Just wanted to validate that this is not an oversight.

Reply:
I moved the special logic for cdsin away from three and made the code exactly same as it used to be before any in delegation